contrib/utilities/indent.py

```diff
@@ -0,0 +1,260 @@
+
+from __future__ import print_function
+
+import argparse
+import time
+import distutils.spawn
+import distutils.file_util
+
+import fnmatch
+import multiprocessing
+import os
+import logging
+import Queue
+import re
+import shutil
+import subprocess
+import threading
+
+from tempfile import mkdtemp, mkstemp
+from filecmp  import cmp
+
+#
+# This script can be used to format code in external projects
+# (that using deal.II) by adding .clang_format files.
+#
+# This program formats code using 'clang-format'.
+# If this program is executed on dry run mode, only the file names of
+# the files that are not formatted correctly are reported
+# without actually formatting them.
+#
+# If this program is executed on dry run mode and if the code base is
+# correctly formatted, then the program doesn't write out anything.
+# This functionality can be exploited by a pre-commit git-hook
+# to check whether the codebase is formatted correctly.
+#
+# Usage:     python contrib/utilities/indent.py
+#
+
+def parse_arguments ():
+  """
+  Argument parser.
+  """
+  parser = argparse.ArgumentParser("Run clang-format on a list of files "
+                                   "in a given list of directories "
+                                   "having a given list of file extensions."
+                                   "This program requires "
+                                   "clang-format version 6.0.")
+
+  parser.add_argument("-b", "--clang-format-binary", metavar="PATH",
+                      default=distutils.spawn.find_executable("clang-format"))
+
+  parser.add_argument("--regex", default="*.cc,*.h,*.cu,*.cuh",
+                      help="Regular expression (regex) to filter files on "
+                      "which clang-format is applied.")
+
+  parser.add_argument("--dry-run", default=False, action='store_true',
+                      help="If --dry-run is passed as an argument, "
+                      "file names of files that are not formatted correctly "
+                      "are written out, without actually formatting them.")
```

Comment:
maybe to add a git command that can set this as a pre-commit hook? Just save users' time from looking.